### PR TITLE
[feature] #2050: Add role-related queries.

### LINKED ISSUE #1986: Add trigger-related queries.
As of writing triggers can't be queried. We need to add at a minimum:
- FindTriggerById
- FindAllTriggers

Optionally also 
- FindTriggersByEvent
Which returns a list of triggers that _would_ get executed for a specific event, but doesn't actually execute them.

### LINKED ISSUE #2050: Add `FindRoleByRoleId` query
`FindRolesByAccountId` returns a `Vec<RoleId>`, so the user either has to use the much more expensive `FindAllRoles` query and manually search for the matching roles, or expect there to be a `FindRoleByRoleId` query which we don't have at the moment.

### PR DESCRIPTION
**Pull request by appetrosyan**
Signed-off-by: Aleksandr Petrosyan <a-p-petrosyan@yandex.ru>

### Description of the Change

Add a few more queries. 

### Issue

Closes #2050 
Closes #1986 

### Benefits

Can now query both roles and triggers.

### Possible Drawbacks

None

### CODE REVIEW

**Review by mversic — DISMISSED** [review]

**Review by mversic — APPROVED** [review]

File: permissions_validators/src/private_blockchain/query.rs [hunk 1]
```diff
@@ -30,7 +30,26 @@ impl<W: WorldTrait> IsAllowed<W, QueryBox> for OnlyAccountsDomain {
                 Err("Only access to the domain of the account is permitted.".to_owned())
             }
             FindAllRoles(_) => Ok(()),
+            FindAllRoleIds(_) => Ok(()),
+            FindRoleByRoleId(_) => Ok(()),
             FindAllPeers(_) => Ok(()),
+            FindAllActiveTriggerIds(_) => Ok(()),
+            FindTriggerById(_) => Ok(()), // TODO: should we allow people to get any trigger they like in a private blockchain also?
+            FindTriggerKeyValueByIdAndKey(query) => {
```
Comment (s8sato) [inline permissions_validators/src/private_blockchain/query.rs:38]: I think all these are world-level access and should return `Err` as `OnlyAccountsDomain` means a restriction to domain-level access

File: permissions_validators/src/private_blockchain/query.rs [hunk 2]
```diff
@@ -261,8 +280,43 @@ impl<W: WorldTrait> IsAllowed<W, QueryBox> for OnlyAccountsData {
...
                 Err("Only access to the assets of the same domain is permitted.".to_owned())
             }
-            FindAllRoles(_) => Ok(()),
-            FindAllPeers(_) => Ok(()),
+            FindAllRoles(_)
+            | FindAllRoleIds(_)
+            | FindRoleByRoleId(_)
+            | FindAllPeers(_)
+            | FindAllActiveTriggerIds(_) => Ok(()),
+            FindTriggerById(query) => {
```
Comment (s8sato) [inline permissions_validators/src/private_blockchain/query.rs:288]: I think all these are world-level access and should return `Err` as `OnlyAccountsData` means a restriction to account-level access

File: permissions_validators/src/private_blockchain/query.rs [hunk 3]
```diff
@@ -261,8 +300,47 @@ impl<W: WorldTrait> IsAllowed<W, QueryBox> for OnlyAccountsData {
...
-            FindAllPeers(_) => Ok(()),
+            FindAllRoles(_)
+            | FindAllRoleIds(_)
+            | FindRoleByRoleId(_)
+            | FindAllActiveTriggerIds(_) => {
+                Err("Only access to the roles of the same account is permitted.".to_owned())
+            }
+            FindAllPeers(_) => {
+                Err("Only access to your account-local data is permitted.".to_owned())
+            }
```
Comment (s8sato) [inline permissions_validators/src/private_blockchain/query.rs:311]:
```suggestion
            FindAllRoles(_) | FindAllRoleIds(_) | FindRoleByRoleId(_) => {
                Err("Only access to the roles of the same account is permitted.".to_owned())
            }
            FindAllActiveTriggerIds(_) | FindAllPeers(_) => {
                Err("Only access to your account-local data is permitted.".to_owned())
            }
```

File: permissions_validators/Cargo.toml [hunk 4]
```diff
@@ -11,6 +11,9 @@ iroha_core = { version = "=2.0.0-pre-rc.4", path = "../core", default-features =
 iroha_data_model = { version = "=2.0.0-pre-rc.4", path = "../data_model", default-features = false }
 iroha_macro = { version = "=2.0.0-pre-rc.4", path = "../macro" }
 iroha_schema = { version = "=2.0.0-pre-rc.4", path = "../schema" }
+iroha_core = { version = "=2.0.0-pre-rc.4", path = "../core", default-features = false }
+iroha_data_model = { version = "=2.0.0-pre-rc.4", path = "../data_model", default-features = false }
+iroha_macro = { version = "=2.0.0-pre-rc.4", path = "../macro" }
```
Comment (s8sato) [inline permissions_validators/Cargo.toml:16]: Duplicated
```suggestion

```

Comment (appetrosyan) [reply permissions_validators/Cargo.toml:16]: I was rebasing before meeting with Vladimir. Rushing never helps.

File: CHANGELOG.rst [hunk 5]
```diff
@@ -761,3 +761,4 @@ Other
   ``proposal_creation_timeout``.
   Update sample config file to have not deprecated DB connection params.
 
+Error: Keyboard Interrupt. Bailing out.
```
Comment (mversic) [inline CHANGELOG.rst:764]: Does this belong in a changelog?

**Review by mversic — DISMISSED** [review]

**Review by s8sato — APPROVED** [review]

**Review by SamHSmith — APPROVED** [review]

**Pull request merged**